### ISSUE TITLE
Strange inconsistency when using --build with conan create

### ISSUE THREAD

**Issue by ytimenkov** [conan-io/conan#4900]
Conan version 1.14.0

I ran `conan create --build=somedep . pkg/1.0@user/test` (where somedep is in pkg's `requires`) to force rebuild of dependency of my package.

somedep was built, but then conan complained:
```
WARN: Can't find a 'pkg/1.0@user/test' package for the specified settings, options and dependencies:
```

Running with `--build somedep --build pkg` worked though, but feels little strange. Maybe `create` should add package-to-create into build list (internally) regardless whether the list is empty or not.

(My intent was to test building 2 packages and then based on json output run `conan remove -b -s` on packages which were built).

**Comment by memsharded** [comment]
Yes, this is a known issue, I have been looking for a previous one, might be a duplicate.

This is a legacy reason, that cannot be fixed until Conan 2.0. It was necessary a way to omit the build of the package when you wanted to test it only, not to build it.

Current command when you define --build is:

```bash
$ conan create . --build=somedep --build=pkg
```

So you need to specify the current package too.

Assigning this to Conan 2.0, that will be when this can be changed.

**Comment by memsharded** [comment]
We have been revisiting this, and our proposal is to keep the current behavior:

- Forcing the implicit ``--build=pkg`` always makes impossible a legit use case, which is not re-building something if there is already a binary for it
- With the package revisions, when the recipe revision is the same, and the package_id is the same, a binary shouldn't be rebuilt. As Conan knows this fact, the ``conan create . --build=missing`` would be a convenient way to avoid unnecessary rebuilds, without losing safety
- The UX design rationale is also something we are trying to have the same criteria in Conan 2.0: Conan provides a default (in this case ``--build=<pkgname>``, if the user wants to provide a different value, it should define it completely. Having hybrids, in which the user writes something and Conan automatically tries to overwrite/modify it, is usually a source of pain.

So lets keep this behavior, thanks for the feedback!
